### PR TITLE
[Drawers.Element] - x/y/defined attributes removed

### PR DESCRIPTION
**Pull request by bluong**
There was filtering done on the Drawer level.  This is now done on the appropriate `XYPlot` level.

### CODE REVIEW

File: src/components/plots/xyPlot.ts [hunk 1]
```diff
@@ -229,8 +216,26 @@ module Plottable {
...
+    }
+
+    protected _getDataToDraw() {
+      var datasets: D3.Map<any[]> = super._getDataToDraw();
+
+      var definedFunction = (d: any, i: number, dataset: Dataset) => {
+        var positionX = Plot._scaledAccessor(this.x())(d, i, dataset);
+        var positionY = Plot._scaledAccessor(this.y())(d, i, dataset);
+        return positionX != null && positionX === positionX &&
+               positionY != null && positionY === positionY;
```
Comment (jtlan) [inline src/components/plots/xyPlot.ts]: Should we use `Utils.isValidNumber()` here?

File: src/components/plots/xyPlot.ts [hunk 2]
```diff
@@ -154,19 +154,6 @@ module Plottable {
       return this;
     }
 
-    protected _propertyProjectors(): AttributeToProjector {
```
Comment (jtlan) [inline src/components/plots/xyPlot.ts]: ??? was this not being used at all?

File: src/components/plots/linePlot.ts [hunk 3]
```diff
@@ -54,10 +54,8 @@ export module Plots {
 
     protected _generateAttrToProjector() {
       var attrToProjector = super._generateAttrToProjector();
-      var wholeDatumAttributes = this._wholeDatumAttributes();
-      var isSingleDatumAttr = (attr: string) => wholeDatumAttributes.indexOf(attr) === -1;
-      var singleDatumAttributes = d3.keys(attrToProjector).filter(isSingleDatumAttr);
-      singleDatumAttributes.forEach((attribute: string) => {
+      d3.keys(attrToProjector).forEach((attribute: string) => {
+        if (attribute === "d") { return; }
```
Comment (jtlan) [inline src/components/plots/linePlot.ts]: Why this change?

File: src/components/plots/linePlot.ts [hunk 4]
```diff
@@ -172,6 +166,14 @@ export module Plots {
                      .defined((innerDatum, innerIndex) => definedProjector(innerDatum, innerIndex, dataset))(datum, index);
       };
     }
+
+    protected _getDataToDraw() {
```
Comment (jtlan) [inline src/components/plots/linePlot.ts]: Why does `Line` have a different implementation for this method?

Comment (bluong) [inline src/components/plots/linePlot.ts]: Because `Line` and `Area` have their own separate way of dealing with undefined data within their appropriate path generators

File: src/components/plots/linePlot.ts (hunk 3 above)
Comment (bluong) [inline src/components/plots/linePlot.ts]: Before the code would loop over anything that is not a `wholeDatumAttribute` which is defined to be x/y/defined/d from `generateAttrToProjector()`

Now that x/y/defined are all gone from `generateAttrToProjector()`, the only one left is d

File: src/components/plots/xyPlot.ts (hunk 1 above)
Comment (bluong) [inline src/components/plots/xyPlot.ts]: Probably.  It'd make the code much more readable

File: src/components/plots/xyPlot.ts (hunk 2 above)
Comment (bluong) [inline src/components/plots/xyPlot.ts]: It was used to set up the xProjector and the yProjector for pieces like `Plots.Bar` and `Drawers.Element`.  Now that the projectors are actually set up correctly, this override can finally go away.
